### PR TITLE
D3ASIM-1145: Added support for calculating unmatched loads incremen…

### PR DESCRIPTION
**Pull request by spyrostz**
…tally, by using only the current market (most recent past market).

### CODE REVIEW

File: src/d3a/models/area/__init__.py [hunk 1]
```diff
@@ -293,7 +293,7 @@ def next_market(self):
         except IndexError:
             return None
 
-    @cached_property
+    @property
```
Comment (hannesdiedrich) [inline src/d3a/models/area/__init__.py]: There are a number of places in the code where @cached_property is used. Why changing it (only) here?

File: integration_tests/steps/jira_issues_tests.py [hunk 2]
```diff
@@ -141,6 +141,7 @@ def pv_produces_same_amount_of_energy_day(context):
                                         for trade in market.trades
                                         if trade.seller == "H2 PV")
                                     for market in same_time_markets]
+        print(same_time_markets_energy)
```
Comment (hannesdiedrich) [inline integration_tests/steps/jira_issues_tests.py]: Please remove. Thanks.

File: integration_tests/steps/integration_tests.py [hunk 3]
```diff
@@ -821,3 +840,17 @@ def assert_trade_rates(context, market_name, trade_rate):
...
+
+@then('the unmatched loads are identical no matter if the past markets are kept')
+def identical_unmatched_loads(context):
+    unmatched_loads = context.simulation.endpoint_buffer.unmatched_loads
+    unmatched_loads_redis = context.simulation.endpoint_buffer.unmatched_loads_redis
+
+    assert len(DeepDiff(unmatched_loads, context.unmatched_loads)) == 0
+    # The 2 simulation runs do not assign the same uuids to the same areas
+    # therefore we have to search whether there are elements with the same values
+
```
Comment (hannesdiedrich) [inline integration_tests/steps/integration_tests.py]: Please move the blank line to the top of this comment (was confused where to connect this comment to) Thanks!

File: src/d3a/models/area/__init__.py (hunk 1 above)
Comment (spyrostz) [reply src/d3a/models/area/__init__.py]: Thanks for pointing it out, I just saw that it was used in other places in the same file, and it probably needs to be removed from there too. The problem was that, due to caching, the current_market property did not always return the last past market, but one-before-the-last at some points. I suspect that the same can happen for the next_market for instance, that's why I think we need to removed the cached_property from all the properties of this file.

File: integration_tests/steps/jira_issues_tests.py (hunk 2 above)
Comment (spyrostz) [reply integration_tests/steps/jira_issues_tests.py]: Thanks, removing

File: integration_tests/steps/integration_tests.py (hunk 3 above)
Comment (spyrostz) [reply integration_tests/steps/integration_tests.py]: Will do, thanks!

File: src/d3a/d3a_core/sim_results/export_unmatched_loads.py [hunk 4]
```diff
@@ -194,3 +212,81 @@ def arrange_output(self, indict, area):
...
+                            if ts not in target_ul[timestamp]['unmatched_times'][device]:
+                                target_ul[timestamp]['unmatched_times'][device].append(ts)
+
+                unm_count = 0
+                for _, hours in target_ul[timestamp]['unmatched_times'].items():
+                    unm_count += len(hours)
+
+                    target_ul['unmatched_count'] = unm_count
+
+    def _merge_accumulated_unmatched_loads(self, accumulated_results, current_results, area):
```
Comment (hannesdiedrich) [inline src/d3a/d3a_core/sim_results/export_unmatched_loads.py]: Just minor, but doesnt this function only copies the value from the current to the accumulated_results? 
The name merge confused me a bit. Again, just personal taste. If you dont agree, just leave it.

File: src/d3a/d3a_core/sim_results/export_unmatched_loads.py [hunk 5]
```diff
@@ -75,20 +85,28 @@ def find_unmatched_loads(self, area, indict):
             self.name_uuid_map[child.name] = child.uuid
             self.name_type_map[child.name] = child.display_type
             if child.children:
-                indict[area.name] = self.find_unmatched_loads(child, indict[area.name])
+                indict[area.name] = self.find_unmatched_loads(
+                    child, indict[area.name], all_past_markets
+                )
             else:
                 if isinstance(child.strategy, LoadHoursStrategy):
+                    curr_market = [child.parent.current_market] \
```
Comment (hannesdiedrich) [inline src/d3a/d3a_core/sim_results/export_unmatched_loads.py]: Sorry for nitpicking, IMO `current_market` would be a better name.

File: src/d3a/d3a_core/sim_results/export_unmatched_loads.py (hunk 4 above)
Comment (spyrostz) [reply src/d3a/d3a_core/sim_results/export_unmatched_loads.py]: Agreedm changing it to copy_accumulated_unmatched_loads

File: src/d3a/d3a_core/sim_results/export_unmatched_loads.py (hunk 5 above)
Comment (spyrostz) [reply src/d3a/d3a_core/sim_results/export_unmatched_loads.py]: agreed, changing that

File: src/d3a/d3a_core/sim_results/export_unmatched_loads.py (hunk 4 above)
Comment (hannesdiedrich) [reply src/d3a/d3a_core/sim_results/export_unmatched_loads.py]: I tried to leave at least very short doc strings, about what the individual functions do in the ExportUnmatchedLoads class. In my experience these can help later when understanding the code, especially if it is for these recursive structures. I suggest to do it also for the MarketUnmatchedLoads. Thanks a lot!

Comment (spyrostz) [reply src/d3a/d3a_core/sim_results/export_unmatched_loads.py]: ok, in that case I will add some docstrings for this class too, thanks!

**Pull request merged**